ambry-cloud/src/main/java/com/github/ambry/cloud/AzureStoreManager.java

```diff
@@ -0,0 +1,237 @@
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.server.ServerErrorCode;
+import com.github.ambry.server.StoreManager;
+import com.github.ambry.store.Store;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Storage manager that tracks {@link AzureStore} objects.
+ */
+public class AzureStoreManager implements StoreManager {
+  private static final Logger logger = LoggerFactory.getLogger(AzureStoreManager.class);
+  protected final ConcurrentHashMap<PartitionId, AzureStore> partitionToAzureStore;
+  protected final MetricRegistry metricRegistry;
+  protected final CloudConfig cloudConfig;
+  protected final AzureCloudConfig azureCloudConfig;
+  protected final AzureMetrics azureMetrics;
+  protected final VcrMetrics vcrMetrics;
+  protected final BlobServiceClient azureStorageClient;
+  protected final VerifiableProperties properties;
+  protected final ClusterMap clusterMap;
+
+  /**
+   * Constructor
+   * @param properties Configuration parameters
+   * @param metricRegistry Metrics
+   * @param clusterMap Cluster-map
+   */
+  public AzureStoreManager(VerifiableProperties properties, MetricRegistry metricRegistry, ClusterMap clusterMap) {
+    this.partitionToAzureStore = new ConcurrentHashMap<>();
+    this.metricRegistry = metricRegistry;
+    this.azureMetrics = new AzureMetrics(metricRegistry);
+    this.vcrMetrics = new VcrMetrics(metricRegistry);
+    this.cloudConfig = new CloudConfig(properties);
+    this.azureCloudConfig = new AzureCloudConfig(properties);
+    this.properties = properties;
+    this.clusterMap = clusterMap;
+    this.azureStorageClient =
+        new ConnectionStringBasedStorageClient(cloudConfig, azureCloudConfig, azureMetrics).getStorageSyncClient();
+    testAzureStorageConnectivity();
+    logger.info("Created AzureStoreManager");
+  }
+
+  /**
+   * Tests connection to Azure blob storage
+   */
+  protected void testAzureStorageConnectivity() {
+    PagedIterable<BlobContainerItem> blobContainerItemPagedIterable = azureStorageClient.listBlobContainers();
+    for (BlobContainerItem blobContainerItem : blobContainerItemPagedIterable) {
+      logger.info("blobContainer = {}", blobContainerItem.getName());
+      break;
+    }
+  }
+
+  /**
+   * Returns blob container in Azure for an Ambry partition
+   * @param partitionId Ambry partition
+   * @return blobContainerClient
+   */
+  protected BlobContainerClient getBlobStore(PartitionId partitionId) {
+    BlobContainerClient blobContainerClient;
+    try {
+      blobContainerClient = azureStorageClient.getBlobContainerClient(String.valueOf(partitionId.getId()));
```

Comment:
Would it be possible to put this Azure container creation logic inside AzureBlobStore (or if possible inside a new class called AzureCloudDestinationV2) similar to how we do today? 

I think today we have:
CloudStorageManager -> CloudBlobStore (contains Ambry logic like retries, encryption, etc) -> CloudDestination / AzureCloudDestination (contains Azure exception handling, retries, etc) -> AzureStorageClient (which uses azure sdk client). 

May be we could have similar to that as below so that Azure interactions are only in  :
CloudStorageManagerV2 -> CloudBlobStoreV2 -> CloudDestination / AzureCloudDestinationV2 -> AzureStorageClient (which uses azure sdk client. (or)

Reply:
Those destination objects in V2 don't really add any value. The code to write to Azure is a 20 lines at most and is best positioned in `AzureStore`

Reply:
If there is no specific reason, if possible, let's be compatible with V1.  
V1 and V2 will coexist for quite some time, it'll improve readability.